django/db/models/expressions.py

```diff
@@ -534,7 +534,7 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
             c.source_expressions[pos] = arg.resolve_expression(query, allow_joins, reuse, summarize, for_save)
         return c
 
-    def as_sql(self, compiler, connection, function=None, template=None):
+    def as_sql(self, compiler, connection, function=None, template=None, arg_joiner=None, **extra_context):
```

Comment:
After looking at #6271, I think I'm seeing the use case for subclasses of `Func`.